dependencies.yaml

```diff
@@ -142,17 +142,19 @@ dependencies:
           - --extra-index-url=https://pypi.nvidia.com
           - --extra-index-url=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple
     specific:
+      # very tight >=x.x.x,<x.x.(x+1) here allows for installation of
+      # post release like 1.15.0.post1
       - output_types: [requirements, pyproject]
         matrices:
           - matrix: {cuda: "12.*"}
             packages:
-              - libucx-cu12==1.15.0
+              - libucx-cu12>=1.15.0,<1.15.1
```

Comment:
Once I removed the `--extra-index-url=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple`, I saw the docs build fail like this:

```text
ERROR: Could not find a version that satisfies the requirement libucx-cu12==1.15.0 (from versions: 1.14.1.post1, 1.15.0.post1, 1.16.0.post1)
ERROR: No matching distribution found for libucx-cu12==1.15.0
```

([build link](https://readthedocs.org/projects/ucx-py/builds/24689587/)

Exactly as reported in #1047.

It looks like a requirement like `==1.15.0` will not match a version like `1.15.0.post` 🤦🏻 

Looked back at a recent successful CI run from a previous job, and I see it was downloading `libucx==1.15.0`, NOT `libucx==1.15.0.post1`.

```text
Downloading https://pypi.anaconda.org/rapidsai-wheels-nightly/simple/libucx-cu11/1.15.0/libucx_cu11-1.15.0-py3-none-manylinux_2_28_x86_64.whl (25.3 MB)
```

([build link](https://github.com/rapidsai/ucx-py/actions/runs/9486356769/job/26140585007))

I think we didn't notice this before because:

* the RAPIDS nightly index is passed as an `--extra-index-url` in CI
* the RAPIDS nightly index contains both `libucx==1.15.0` and `libucx==1.15.0.post1`

But we're now noticing it in the docs build because:

* pypi.org does not have `libucx==1.15.0`, only `libucx==1.15.0.post1`

How I tested that this new specifier would work:

```text
pip install --dry-run 'libucx-cu12==1.15.0'
# ERROR: Could not find a version that satisfies the requirement libucx-cu12==1.15.0 (from versions: 1.14.1.post1, 1.15.0.post1, 1.16.0.post1)

pip install --dry-run 'libucx-cu12>=1.15.0,<1.15.1'
# Would install libucx-cu12-1.15.0.post1
```

cc @vyasr